site/source/docs/porting/exceptions.rst

```diff
@@ -0,0 +1,50 @@
+.. Exceptions support:
+
+==============================
+C++ exceptions support
+==============================
+
+C++ exceptions support can be enabled in emscripten using ``-fexceptions``.
+Note that unlike on most platforms, C++ exceptions are disabled by default due
+to their current overhead, see later.
+
+Passing ``-fexceptions`` during compilation will pass the relevant flags to
+clang, and during link it also links in system libraries with exceptions
+support.
+
+Example
+#######
+
+For example, build the following program with ``emcc -fexceptions``:
+
+.. code-block:: cpp
+
+    #include <stdio.h>
+
+    int main() {
+      try {
+        puts("throw...");
+        throw 1;
+        puts("(never reached)");
+      } catch(...) {
+        puts("catch!");
+      }
+      return 0;
+    }
+
+Running it will print ``throw...`` and then ``catch!``.
+
+Overhead
+########
+
+Until wasm gets direct support for exceptions, Emscripten supports them using
+JavaScript for throwing and catching. As a result, every time we might need to
+catch an exception we call out into JavaScript to a try-catch there. This
+increases code size and adds significant overhead.
+
+You can reduce the overhead by specifying a list of allowed functions in
+which exceptions are enabled, see the ``EXCEPTION_CATCHING_ALLOWED`` setting.
+
+In the future direct support for exceptions in wasm will avoid that overhead.
+Usage will be as mentioned earlier, but the exceptions feature will also need to
+be enabled when compiling, using ``-fwasm-exceptions``.
```

Comment:
Won't `-fwasm-exceptions` replace `-fexceptions`? This paragraph makes it sound like you will need both. This is unlike other features, which are enabled by `-m` flags that do not affect program semantics.

Reply:
The current status is if you use `-fexceptions` it uses Emscripten EH and if you use `-fwasm-exceptions` it uses wasm EH. My long term plan is making `-fexceptions` the same as `-fwasm-exceptions`, but it should be after all major browsers ship EH and it is enabled by default without needing to use any flags, so it will be a long way.

Currently when using wasm EH you only need `-fwasm-exceptions` and don't need to use both flags.

Reply:
Thanks, fixed to mention just the one flag will be needed.